homeassistant/components/tplink/entity.py

```diff
@@ -279,16 +286,27 @@ def _description_for_feature[_D: EntityDescription](
             # _attr_name > translation.name > description.name
             # > device_class (if base platform supports).
             entity_name: str | None | UndefinedType = UNDEFINED
+            has_entity_name = True
 
             # The state feature gets the device name or the child device
             # name if it's a child device
             if feature.id == PRIMARY_STATE_ID:
                 translation_key = None
-                entity_name = child_alias  # if None will use device name
+                entity_name = (
+                    get_device_name(device) if parent else None
+                )  # if None will use device name
```

Comment:
Move the comment above the line, this will make the code itself a oneliner.

Reply:
Done